examples/basic_shared_vpc/main.tf

```
@@ -57,7 +57,7 @@ resource "google_compute_address" "internal" {
 resource "google_compute_instance" "reserved_ip" {
   project      = var.service_project
   zone         = "us-central1-a"
-  name         = google_compute_address.internal.self_link
+  name         = "reserved-ip"
```

Just to make to clearer that this is a VM resource.

```suggestion
  name         = "reserved-ip-instance"
```